LDAR_Sim/src/virtual_world/sources.py

```diff
@@ -54,6 +55,76 @@ def __init__(self, id: str, info, prop_params) -> None:
         self._set_source_properties(prop_params=prop_params)
         self._next_emission: Emission = None
 
+    def __reduce__(self) -> str | tuple[Any, ...]:
+        if self._repairable:
+            args = (
+                self._source_ID,
+                self._repairable,
+                self._persistent,
+                self._active_duration,
+                self._inactive_duration,
+                self._generated_emissions,
+                self._emis_rate_source,
+                self._emis_prod_rate,
+                self._emis_duration,
+                self._meth_spat_covs,
+                self._emis_rep_delay,
```

Comment:
Question: What do you do with rep delay and rep cost if it's not repairable?

Reply:
when it's not repairable, those two values are reduced to None values. So when the object is reconstructed, these are set to None.